### PR TITLE
Integrate docker support to EVA

### PR DESCRIPTION
**Pull request by jiashenC**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    && ~/miniconda/bin/conda info -a
+
+COPY . /app
+RUN sudo chown -R eva:eva /app
+
+# install eva env
+RUN ~/miniconda/bin/conda env create -f /app/script/install/conda_eva_environment.yml
+
+# sanity check for GPU
+RUN ~/miniconda/bin/conda run -n eva python -c "import torch; torch.randn(1).cuda()"
```
Comment (gaurav274) [inline docker/Dockerfile:31]:
```
ERROR conda.cli.main_run:execute(31): Subprocess for 'conda run ['python', '-c', 'import torch; torch.randn(1).cuda()']' command failed.  (See above for error)
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/eva/miniconda/envs/eva/lib/python3.7/site-packages/torch/cuda/_init_.py", line 192, in _lazy_init
    _check_driver()
  File "/home/eva/miniconda/envs/eva/lib/python3.7/site-packages/torch/cuda/_init_.py", line 102, in _check_driver
    http://www.nvidia.com/Download/index.aspx""")
AssertionError: 
Found no NVIDIA driver on your system. Please check that you
have an NVIDIA GPU and installed a driver from
http://www.nvidia.com/Download/index.aspx
```
Encountered above error when tested on ada1

**Review by gaurav274 — CHANGES_REQUESTED** [review]
[Fix Required] Docker is unable to detect CUDA

**Comment by jiashenC** [comment]
Going to investigate. 

It would be better if we can find some clean machines for running the GPU related CI, so we know the exact per-requisite packages.

File: docker/Dockerfile (hunk 1 above)
Comment (jiashenC) [reply docker/Dockerfile]: Add a CPU only test case to bypass the GPU dependency. 
https://github.com/georgia-tech-db/eva/pull/103/files#diff-4e5e90c6228fd48698d074241c2ba760R32

File: docker/eva_gpu.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    && ~/miniconda/bin/conda update -q conda \
+    && ~/miniconda/bin/conda info -a
+
+COPY . /app
+
+# install eva env
+RUN ~/miniconda/bin/conda env create -f /app/script/install/conda_eva_environment.yml
+
+# sanity check for GPU
+RUN ~/miniconda/bin/conda run -n eva python -c "import torch; torch.randn(1).cuda()"
```
Comment (gaurav274) [inline docker/eva_gpu.Dockerfile:27]: GPU image failed. We need to expose GPU drivers to the docker.

**Review by gaurav274 — APPROVED** [review]
I think we can merge the PR and open an issue for the GPU image.

File: README.md [hunk 3]
```diff
@@ -38,6 +39,13 @@ export PATH="$HOME/miniconda/bin:$PATH"
 sh script/install/install.sh
 ```
 
+## Docker
+
+1. Standup EVA testing for CPU/GPU hardware.
+```shell
+docker-compose -f docker-compose.yml eva-test-[cpu/gpu]
+```
```
[the remark this answers is not in the document]
Comment (jiashenC) [reply README.md:50]: Good catch LOL.

File: README.md [hunk 4]
```diff
@@ -38,6 +39,13 @@ export PATH="$HOME/miniconda/bin:$PATH"
 sh script/install/install.sh
 ```
 
+## Docker
+
+1. Standup EVA testing for CPU/GPU hardware.
+```shell
```
[the remark this answers is not in the document]
Comment (jiashenC) [reply README.md:48]: Add doc link.

**Pull request merged**